File: output/gemini_3_1_pro_preview/zero/python/Fita Colorida.py
import sys

def solve():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    
    N = int(input_data[0])
    fita = list(map(int, input_data[1:N+1]))
    
    dist = [float('inf')] * N
    
    last_zero = float('-inf')
    for i in range(N):
        if fita[i] == 0:
            last_zero = i
        if last_zero != float('-inf'):
            dist[i] = i - last_zero
            
    last_zero = float('inf')
    for i in range(N - 1, -1, -1):
        if fita[i] == 0:
            last_zero = i
        if last_zero != float('inf'):
            dist[i] = min(dist[i], last_zero - i)
            
    for i in range(N):
        if dist[i] >= 9:
            dist[i] = 9
            
    print(*(dist))

if __name__ == '__main__':
    solve()

File: output/csbc_semish_2026/deepseek_deepseek_v3_2/zero/cpp/Fita Colorida.cpp

#include <iostream>
#include <vector>
#include <algorithm>
#include <climits>

using namespace std;

int main() {
    int N;
    cin >> N;
    
    vector<int> fita(N);
    for (int i = 0; i < N; i++) {
        cin >> fita[i];
    }
    
    vector<int> dist(N, INT_MAX);
    
    int lastZero = -1;
    for (int i = 0; i < N; i++) {
        if (fita[i] == 0) {
            lastZero = i;
        }
        if (lastZero != -1) {
            dist[i] = min(dist[i], i - lastZero);
        }
    }
    
    lastZero = -1;
    for (int i = N - 1; i >= 0; i--) {
        if (fita[i] == 0) {
            lastZero = i;
        }
        if (lastZero != -1) {
            dist[i] = min(dist[i], lastZero - i);
        }
    }
    
    for (int i = 0; i < N; i++) {
        if (dist[i] > 9) {
            dist[i] = 9;
        }
        cout << dist[i];
        if (i < N - 1) {
            cout << " ";
        }
    }
    cout << endl;
    
    return 0;
}
